contracts/ExecutionNodeVersionBeacon.cdc

```diff
@@ -0,0 +1,425 @@
+    access(contract) var versionUpdateBuffer: UInt64
+    /// Set expectations regarding how much the versionUpdateBuffer can vary between version boundaries
+    access(contract) var versionUpdateBufferVariance: UFix64
+
+    /// Sorted Array containing future block heights where a version boundary is defined
+    /// We'll maintain this as easy mechanism for determining proximal version block boundaries
+    access(contract) var upcomingBlockBoundaries: [UInt64]
+
+    /// Sorted Array containing historical block heights where version boundaries were defined
+    access(contract) var archivedBlockBoundaries: [UInt64]
+
+    /// Admin interface that manages the Execution Node versioning
+    /// maintained in this contract
+    pub resource interface ExecutionNodeVersionAdmin {
+
+        /// Update the minimum version to take effect at the given block height
+        pub fun addVersionBoundaryToTable(targetBlockHeight: UInt64, newVersion: Semver) {
+            pre {
+                targetBlockHeight > getCurrentBlock().height + ExecutionNodeVersionBeacon.versionUpdateBuffer
+                    : "Target block height occurs too soon to update EN version."
+            }
+        }
+
+        /// Deletes the last entry in versionTable which defines an upcoming version boundary
+        /// Could be helpful in case rollback is needed
+        pub fun deleteUpcomingVersionBoundary(blockHeight: UInt64): Semver {
+            pre {
+                ExecutionNodeVersionBeacon.versionTable.length > 0: "No version boundary mappings exist."
+                ExecutionNodeVersionBeacon.versionTable.keys.contains(blockHeight): "No boundary defined at that blockHeight."
+                ExecutionNodeVersionBeacon.upcomingBlockBoundaries.contains(blockHeight): "Boundary not defined in upcomingBlockBoundaries."
+            }
+        }
+
+        /// Updates the number of blocks that must buffer updates to the versionTable
+        /// and the block number the version is targetting
+        pub fun changeVersionUpdateBuffer(newUpdateBufferInBlocks: UInt64) {
+            post {
+                ExecutionNodeVersionBeacon.versionUpdateBuffer == newUpdateBufferInBlocks: "Update buffer was not properly changed! Reverted."
+            }
+        }
+
+        /// Updates the variance percentage by which the versionUpdateBuffer can be changed
+        /// Value must be between 0.0 and 1.0
+        pub fun changeVersionUpdateBufferVariance(_ newUpdateBufferVariance: UFix64) {
+            pre {
+                newUpdateBufferVariance >= 0.0 && newUpdateBufferVariance <= 1.0 : "Buffer variance must be value between 0 and 1."
+            }
+            post {
+                ExecutionNodeVersionBeacon.versionUpdateBufferVariance == newUpdateBufferVariance: "Update buffer variance was not properly changed! Reverted."
+            }
+        }
+    }
+
+    /// Admin resource that manages the Execution Node versioning
+    /// maintained in this contract
```

Comment:
This is the same comment that is above the previous resource interface definition. Which one is it supposed to be before?